src/server/qtquick/wsurfaceitem.cpp

```diff
@@ -210,7 +210,7 @@ class Q_DECL_HIDDEN WSurfaceItemContentPrivate: public QQuickItemPrivate
 
         // wayland protocol job should not run in rendering thread, so set context qobject to contentItem
         frameDoneConnection = QObject::connect(q->window(), &QQuickWindow::afterRendering, q, [this, q](){
-            if ((q->rendered || q->isVisible()) && live && surface) {
+            if ((q->rendered || q->isVisible()) && live && surface && !surface->isInvalidated()) {
```

Comment:
Needs handle `aboutToBeInvalidated` signal to ensure the surface is valid at here.